fedimint-server/src/lib.rs

```diff
@@ -293,3 +293,13 @@ impl FedimintApiHandler {
         self.handle.stopped().await;
     }
 }
+
+pub type ApiResult<T> = std::result::Result<T, ApiError>;
+
+pub fn check_auth(context: &mut ApiEndpointContext) -> ApiResult<()> {
```

Comment:
I wonder if this could be a method of `ApiEndpointContext`.

Reply:
Or would it fit as an additional macro? `api_endpoint!` and then `api_endpoint_with_auth!`